task/control-plane-agent/src/main.rs

```diff
@@ -155,6 +155,23 @@ enum MgsMessage {
 
 ringbuf!(Log, 16, Log::Empty);
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum CriticalEvent {
+    Empty,
+    /// We have received a network request to change power states. This record
+    /// logs the sender, in case the request was unexpected, and the target
+    /// state.
+    SetPowerState {
+        sender: sp_impl::SocketAddrV6,
+        port: SpPort,
+        power_state: PowerState,
```

Comment:
Do you want to include a "timestamp" here too (probably system ticks)?

Reply:
If we do that, it seems potentially valuable to add the same timestamp to the non-CRITICAL ringbuf events, so that the temporal relationship between the CRITICAL and non-CRITICAL events could be correlated...honestly, even just a sequence number could do that...